src/nvim/testdir/test_eval.in

```diff
@@ -122,7 +121,19 @@ call SetReg('/', ['abc/'])
 call SetReg('/', ["abc/\n"])
 call SetReg('=', ['"abc/"'])
 call SetReg('=', ["\"abc/\n\""])
-
+$put ='{{{1 System clipboard'
+" Save and restore system clipboard.
+" If no connection to X-Server is possible, test should succeed.
+:let _clipreg = ['+', getreg('+'), getregtype('+')]
+:let _clipopt = &cb
+:let &cb='unnamedplus'
+:1y
+:AR +
+:tabdo :windo :echo "hi"
```

Comment:
I think it was a bug in vim that `:tabdo :windo` would call `begin_`/`end_global_changes` recursively before that was supported, fixed by 598 . Do you mean that rtp is setup in the oldtests so that `autoload/provider/clipboard.vim` is not in `rtp` for the tests ? That would be fixed anyway by converting to lua test and use the shim provider.